infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -309,6 +364,46 @@ private boolean runRecoveryReconfiguration() {
         return recoveryResult;
     }
 
+    /**
+     * Task to collect local node metrics.
+     * This task collects the following:
+     * Boolean Status of layout, sequencer and logunit servers.
+     * Logunit Metrics:
+     * Size of writeQueue for pending write requests.
+     * These metrics are then composed into NodeMetrics protobuf model
+     * and stored locally.
+     */
+    private void updateLocalMetrics() {
+        // Initializing the status of components
+        // No need to poll unless node is bootstrapped.
+        Layout layout = serverContext.getManagementLayout();
+        if (layout == null) {
+            return;
+        }
+
+        // Build and replace existing locally stored nodeMetrics
+        SequencerMetrics sequencerMetrics;
+        if (layout.getSequencers().get(0).equals(getLocalEndpoint())) {
+            try {
+                sequencerMetrics = CFUtils.getUninterruptibly(
+                        getCorfuRuntime()
+                                .getLayoutView().getRuntimeLayout(layout)
+                                .getSequencerClient(getLocalEndpoint())
+                                .requestMetrics());
+            } catch (ServerNotReadyException snre) {
+                sequencerMetrics = new SequencerMetrics(SequencerStatus.NOT_READY);
+            } catch (Exception e) {
+                log.error("Error while requesting metrics from the sequencer: ", e);
+                sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
+            }
+        } else {
+            sequencerMetrics = new SequencerMetrics(SequencerStatus.UNKNOWN);
```

Comment:
Right, but this status is reported by the sequencer and then later analyzed by the management server.
The sequencer, however, cannot realize that it is a primary/non-primary sequencer as it has no view of the layout. All it can say whether is ready or not.

Reply:
but this is neither that it is ready or not ready, it is actually that no sequencer exists on the current endpoint, right? so technically there is no such thing as being ready or not, as it is not found in the layout.